### ISSUE TITLE
Proxy before containers = no-go

### ISSUE THREAD

**Issue by taiidani** [gesellix/docker-haproxy-network#1]
In the case of containers being routed to by HAProxy that are not available at startup, HAProxy will terminate out of the gate. Ideal state would be that HAProxy marks that backend as offline (502 error) and when its container comes online registers it and brings it online.

I've searched far and wide for an HAProxy setting that would disable this startup health check and/or allow for initial DNS resolution to fail cleanly. No go though.

Do you happen to know how to deal with this behavior? I've spun up a (friendly competition) [separate repository](https://github.com/taiidani/docker-nginx-reverse-proxy) that implements the behavior with Nginx but would love to see an HAProxy implementation.

**Comment by gesellix** [comment]
Hey, sorry for the delay.

You're right that HAProxy will fail to startup when the dns resolve fails. Digging through some forum threads it seems like they won't fix it. The main issue is the definition of backend dns names: they are expected to be _resolvable_. Non-resolvable is considered to be an error.

So, using alternatives like nginx or traefik seems to be the only way to get some kind of guaranteed startup. Another option obviously is to use other ways (scripts, management tools) to ensure resolvable backends before starting the HAProxy.

**Comment by gesellix** [comment]
P.S.: thanks for the friendly competition, I added a link to your repo in the readme ;-)
